src/Security/Authentication/Handler/ProcessSamlAuthenticationHandler.php

```diff
@@ -0,0 +1,66 @@
+<?php declare(strict_types=1);
+
+/**
+ * Copyright 2023 SURFnet B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Surfnet\SamlBundle\Security\Authentication\Handler;
+
+use Exception;
+use SAML2\Assertion;
+use SAML2\Response\Exception\PreconditionNotMetException;
+use Surfnet\SamlBundle\Http\Exception\AuthnFailedSamlResponseException;
+use Surfnet\SamlBundle\Monolog\SamlAuthenticationLogger;
+use Surfnet\SamlBundle\SAML2\Response\Assertion\InResponseTo;
+use Surfnet\SamlBundle\Security\Authentication\SamlAuthenticationStateHandler;
+use Surfnet\SamlBundle\Security\Authentication\SamlInteractionProvider;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\Security\Core\Exception\AuthenticationException;
+
+class ProcessSamlAuthenticationHandler implements AuthenticationHandler
+{
+    public function __construct(
+        private readonly SamlInteractionProvider $samlInteractionProvider,
```

Comment:
PHP 8.1 specific code where this lib is supposed to run on PHP 8.0?

Reply:
Very good point! Removing them from the project for now